sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -523,13 +524,14 @@ class SessionCatalog(
    */
   def dropPartitions(
       tableName: TableIdentifier,
-      parts: Seq[TablePartitionSpec],
+      specs: Seq[TablePartitionSpec],
       ignoreIfNotExists: Boolean): Unit = {
+    requirePartialMatchedPartitionSpec(specs, getTableMetadata(tableName))
```

Comment:
Thus, this PR resolves the issue by banning users to specify the invalid partition spec. 

Below is the test case:
https://github.com/gatorsmile/spark/blob/9ca76217defe5e0154b1d26a61baf4831d19df3a/sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala#L189-L194

Without this PR fix, it will drop all the partitions.